File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-0.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> st;

        for(string& token : tokens){
            if(token != "-" && token != "+" && token != "*" && token != "/"){
                st.push(stoi(token));
            } else {
                int b = st.top(); st.pop();
                int a = st.top(); st.pop();

                int results = 0;
                if (token == "+"){
                results = a+ b;
                } else if (token == "-"){
                    results = a - b;
                } else if(token == "*"){
                    results = a*b;
                } else{
                    results = a/b;
                }

                st.push(results);
            }
        }  
        return st.top();  
    }
};

File: Data Structures & Algorithms/validate-parentheses/submission-0.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        pMap = {")" : "(", "]":"[", "}": "{"}
        for char in s:
            if char in pMap:
                top = stack.pop() if stack else "#"
                if pMap[char] != top:
                    return False
            else:
                stack.append(char)
        return not stack
